WorkflowTesting/Sources/WorkflowRenderTester.swift

```diff
@@ -253,18 +253,27 @@
         }
     }
 
-    extension Collection {
-        fileprivate func appending(_ element: Element) -> [Element] {
+    fileprivate extension Collection {
+        func appending(_ element: Element) -> [Element] {
             return self + [element]
         }
     }
 
-    extension Dictionary {
-        fileprivate func setting(key: Key, value: Value) -> [Key: Value] {
+    fileprivate extension Dictionary {
+        func setting(key: Key, value: Value) -> [Key: Value] {
             var newDictionary = self
             newDictionary[key] = value
             return newDictionary
         }
     }
 
+    public extension RenderTester {
+        /// Convenience to add expectation for `RenderLatestOutputWorkflow`
+        /// - Parameters:
+        ///   - producing: Value to return as `Rendering`
+        ///   - key: The key of the expected workflow (if specified).
+        func expectLatestOutputRenderingWorkflow<Value>(producing: Value, key: String = "") -> RenderTester<WorkflowType> {
```

Comment:
Thinking about this more, I’m wondering if it would be nicer for the test writer to not need to be concerned with the workflow implementation of this. Imagine a workflow like:

```swift
extension AccountLoadingWorkflow {

    func render(state: State, context: RenderContext<Self>) -> AccountLoadingScreen {
        let loadState = AccountLoadingWorker(id: id)
            .mapOutput { result -> LoadState<Account, Error> in
                switch result {
                case .success(let account):
                    return .loaded(account)
                case .failure(let error):
                    return .failed(error)
                }
            }
            .renderLatestOutput(startingWith: .loading)

        return AccountLoadingScreen(state: loadState)
    }

}
```

It’d be nice to have a test like:

```swift
AccountLoadingWorkflow(id: "b3n")
    .renderTester()
    .expect(worker: AccountLoadingWorker(id: "b3n"))
    .render { screen in
        XCTAssertEqual(screen.state, .loading)
    }
```

This lets me test that the underlying worker is actually configured as expected and leaves this feeling like an “operator” rather than another layer in the tree. I’m not sure how we’d handle non-initial values with that method, however. (I’m also not _entirely_ sure how we’d pull this off exactly, but I think we could _probably_ find a way?)

Reply:
Yes, this would've been great. This was my first thought, I dug around a bit, but didn't get too far with it.

The current `AnyWorkflow` operators do output/rendering mappings, I wasn't able to figure out a way to project outputs to renderings. If we could do something like that, we would be able to write some powerful operators (like `merge` multiple Workflows...etc).